site/src/components/AuditLogRow/AuditLogRow.tsx

```diff
@@ -16,7 +16,28 @@ import userAgentParser from "ua-parser-js"
 import { combineClasses } from "util/combineClasses"
 import { AuditLogDiff } from "./AuditLogDiff"
 
-const readableActionMessage = (auditLog: AuditLog) => {
+// the BE returns additional_field as a string, since it's stored as JSON but
+// technically, it's a map, so we adjust the type here.
+type ExtendedAuditLog = Omit<AuditLog, "additional_fields"> & {
+  additional_fields: Record<string, string>
+}
```

Comment:
Not proud of this lol
I am not sure how to amend the BE type - the DB needs it stored as JSON which seems to correspond to string when `make gen` is run.  Open to suggestions.

Reply:
Not totally sure but gonna check it out!  This seems like a reasonable workaround to me for the time being though.

Reply:
So we could get closer by adding:
 
```
		case "encoding/json.RawMessage":
			return TypescriptType{ValueType: "object"}, nil
```

to `scripts/apitypings/main.go:691`.  But that gets us `object` which we would still need to typecast (better than `string` though).

To fully resolve I think we might have to type the thing instead of using `json.RawMessage` and make every property optional.

Or we have to add separate audit log types for every type of audit log that has additional fields and on the frontend we check which it is with some kind of `function (a: AuditLog) => a is WorkspaceBuildAuditLog` type of deal.

I feel like making it an object and casting is pretty decent for now though.

@Emyrk just in case you have some insight

Reply:
Er wait if `Record<string, string> works without typecasting we can just do:
```
		case "encoding/json.RawMessage":
			return TypescriptType{ValueType: "Record<string, string>"}, nil
```